socialhome/streams/app/stores/streamStore.js

```diff
@@ -10,15 +11,65 @@ import {actions, mutations, streamStoreOperations, getters} from "streams/app/st
 
 Vue.use(Vuex)
 
+function newRestAPI(options) {
+    const opts = _defaults({}, options, {
+        baseURL: "/api",
+        axios: Vue.prototype.$http,
+    })
+
+    const onSuccess = (state, payload) => {
+        payload.data.forEach(item => {
+            Vue.set(state.contents, item.id, item)
+            state.contentIds.push(item.id)
+        })
+    }
+
+    const onError = (state, error) => {
+        /* TODO: Proper error handling */
+        console.error(`An error happened while fetching post: ${error}`)
+    }
+
+    return new Vapi(opts)
+        .get({
+            action: "getPublicStream",
+            path: "/streams/public",
```

Comment:
Since we have the JS urls available, we should use those. For example, `api:streams-public` for this one, if I'm not mistaken. All the other urls here should also do the same.